contracts/BaseLaunchpeg.sol

```diff
@@ -360,6 +431,76 @@ abstract contract BaseLaunchpeg is
         emit DevMint(msg.sender, _quantity);
     }
 
+    /// @dev Should only be called in the pre-mint phase
+    /// @param _quantity Quantity of NFTS to mint
+    function _preMint(uint256 _quantity) internal {
+        if (_quantity == 0) {
+            revert Launchpeg__InvalidQuantity();
+        }
+        if (_quantity > allowlist[msg.sender]) {
+            revert Launchpeg__NotEligibleForAllowlistMint();
+        }
+        if (
+            totalSupply() + _quantity > collectionSize ||
+            amountMintedDuringAllowlist + _quantity > amountForAllowlist
+        ) {
+            revert Launchpeg__MaxSupplyReached();
+        }
+        allowlist[msg.sender] -= _quantity;
+        uint256 price = _preMintPrice();
+        uint256 totalCost = price * _quantity;
+        numberPreMinted[msg.sender] += _quantity;
+        amountMintedDuringPreMint += _quantity;
+        amountMintedDuringAllowlist += _quantity;
+        preMintQueue.push(
+            PreMintData({sender: msg.sender, quantity: _quantity})
+        );
+        emit PreMint(msg.sender, _quantity, price);
+        _refundIfOver(totalCost);
+    }
+
+    /// @dev Should only be called in the allowlist and public sale phases.
+    /// @param _maxQuantity Max quantity of NFTs to mint
+    function _batchMint(uint256 _maxQuantity) internal {
```

Comment:
nit: it's not clear to me that `batchMint` is a function to mint NFTs reserved during the pre-mint phase

we could call it `batchMintPreMintedNFTs` or something similar to avoid confusion, what do you think?

Reply:
@0xmurloc do you have any suggestions/views on calling it `batchMintPreMintedNFTs` instead?